src/Tests/Microsoft.NET.Build.Containers.IntegrationTests/TargetsTests.cs

```diff
@@ -31,6 +31,21 @@ public void CanSetEntrypointArgsToUseAppHost(bool useAppHost, params string[] en
         }
     }
 
+    [Fact]
+    public void CanDeferToContainerImageNameWhenPresent() {
+        var customImageName = "my-container-app";
+        var (project, logger, d) = ProjectInitializer.InitProject(new()
+        {
+            [ContainerImageName] = customImageName
+        });
+        using var _ = d;
+        var instance = project.CreateProjectInstance(global::Microsoft.Build.Execution.ProjectInstanceSettings.None);
+        instance.Build(new[] { ComputeContainerConfig }, new []{ logger });
+        logger.Warnings.Should().HaveCount(1, "a warning for the use of the old ContainerImageName property should have been created");
+        logger.Warnings[0].Code.Should().Be(KnownStrings.ErrorCodes.CONTAINER003);
+        Assert.Equal(customImageName, instance.GetPropertyValue(ContainerRepository));
+    }
```

Comment:
@vlada-shubina here is the test that confirms that we still use the previous property - just with a warning that lets users know to migrate